### DISCUSSION TITLE
Fast enough?

### DISCUSSION THREAD

**Discussion by wocard2** [ataradov/usb-sniffer#8] [General]
Is this design (using HS USB for transferring the captured data) really fast enough for capturing HS USB data? As a header is added to the data, there's more data transferred from the sniffer to the host than data captured and  there will be cases where data is lost.

As the FX3 (USB3 SuperSpeed) chips cannot be hand-soldered (BGA!), a daughterboard for the CYUSB3KIT-003 could be a solution.

**Comment by ataradov** [comment]
It is good enough in practice. In theory you can overflow and cause some packets to drop. This would be noted in the log. But this is not a real issue because by that time there will be so many packets that Wireshark and capture pipe interface may be the bottleneck anyway.

I captured transfer of large files from a USB drive without any issues. And I could not find anything that would intentionally overflow the buffer.

Most practical devices in combination with drivers do not utilize full bandwidth during normal operation.

**Comment by wocard2** [comment]
Thanks!

Oops, the CYUSB3KIT-003 is no longer available and the CYUSB3KIT-004 is quite expensive.

**Comment by ataradov** [comment]
And also, the most common use for a tool like this is not debugging the transfer of huge volumes of data, but debugging the basic logic. You usually setup a scenario where there is as little data as possible to debug your issue.
